arangod/IResearch/IResearchFeature.cpp

```diff
@@ -163,6 +165,42 @@ void registerRecoveryHelper() {
   }
 }
 
+void transactionStateRegistrationCallback(
+    TRI_voc_cid_t cid,
+    arangodb::TransactionState& state
+) {
+  auto* vocbase = state.vocbase();
+
+  if (!vocbase) {
+    LOG_TOPIC(WARN, arangodb::iresearch::IResearchFeature::IRESEARCH)
```

Comment:
The caller will never get to know if something went wrong during callback registration. 
Wouldn't it be better to return something from this function to show registration was successful?

Reply:
it's valid for transactionStateRegistrationCallback(...) to not do anything if the 'cid' did not match its requirements, e.g. a view callback was passed a collection cid

Reply:
however, if the same API is going to be used to delegate collection registration to MMFilesTransactionState/RocksDBTransactionState then a return code can be used to abort collection addition
done